pkg/resourcecollector/resourcecollector.go

```diff
@@ -92,18 +106,22 @@ func resourceToBeCollected(resource metav1.APIResource) bool {
 }
 
 // GetResources gets all the resources in the given list of namespaces which match the labelSelectors
-func (r *ResourceCollector) GetResources(namespaces []string, labelSelectors map[string]string) ([]runtime.Unstructured, error) {
-	err := r.discoveryHelper.Refresh()
+func (r *ResourceCollector) GetResources(namespaces []string, labelSelectors map[string]string, remoteConfig *restclient.Config) ([]runtime.Unstructured, error) {
+	err := r.initClusterConfig(remoteConfig)
```

Comment:
We can't do this either, the same instance of `ResourceCollector` is being used by different modules. You need to call `k8s.NewInstanceFromRestConfig()` from here and use the sched-ops instance instead of setting it as `r.k8sOps`